### PR TITLE
ddl: fix 'create table like' for partitioned table

### PR DESCRIPTION
**Pull request by tiancaiamao**
### What problem does this PR solve? 

Fix https://github.com/pingcap/tidb/issues/10784

### What is changed and how it works?

"create table like" should not shadow copy the partition info from the original table, it should use new tableID.

`genGlobal` is updated so we can get a batch of tableID in one call

### Check List 

Tests 

 - Unit test

Related changes

 - Need to cherry-pick to the release branch

### CODE REVIEW

**Comment by tiancaiamao** [comment]
PTAL @jackysp @crazycs520 @zimulala

File: ddl/ddl.go [hunk 1]
```diff
@@ -495,8 +495,8 @@ func (d *ddl) GetInfoSchemaWithInterceptor(ctx sessionctx.Context) infoschema.In
 	return d.mu.interceptor.OnGetInfoSchema(ctx, is)
 }
 
-func (d *ddl) genGlobalID() (int64, error) {
-	var globalID int64
+func (d *ddl) genGlobalID(count int) ([]int64, error) {
```
Comment (zimulala) [inline ddl/ddl.go]: s/genGlobalID/genGlobalIDs ?

Comment (tiancaiamao) [reply ddl/ddl.go]: Done

**Pull request merged**